arrows/mvg/applets/init_cameras_landmarks.cxx

```diff
@@ -0,0 +1,644 @@
+        if (all_lms.find(id) == all_lms.end())
+        {
+          all_lms[id] = kwiver::vital::landmark_sptr();
+        }
+      }
+      landmark_map_ptr =
+        std::make_shared<kwiver::vital::simple_landmark_map>(all_lms);
+    }
+
+    algorithm->initialize(camera_map_ptr, landmark_map_ptr,
+                          feature_track_set_ptr, sfm_constraint_ptr);
+  }
+
+  std::string get_filename( kwiver::vital::frame_id_t frame_id )
+  {
+
+    if (sfm_constraint_ptr && sfm_constraint_ptr->get_metadata())
+    {
+      auto videoMetadataMap = sfm_constraint_ptr->get_metadata();
+      auto mdv = videoMetadataMap->get_vector(frame_id);
+      if (!mdv.empty())
+      {
+        return frameName(frame_id, mdv);
+      }
+    }
+    auto dummy_md = std::make_shared<kwiver::vital::metadata>();
+    dummy_md->add<kwiver::vital::VITAL_META_VIDEO_URI>(std::string(video_name));
+    return frameName(frame_id, dummy_md);
+  }
+};
+
+// ----------------------------------------------------------------------------
+int
+init_cameras_landmarks::
+run()
+{
+  try
+  {
+    switch(d->process_command_line(command_args()))
+    {
+      case priv::HELP:
+        std::cout << m_cmd_options->help();
+        return EXIT_SUCCESS;
+      case priv::WRITE:
+        return EXIT_SUCCESS;
+      case priv::FAIL:
+        return EXIT_FAILURE;
+      case priv::SUCCESS:
+        ;
+    }
+
+    if ( d->config == nullptr )
+    {
+      return EXIT_FAILURE;
+    }
+
+    if(d->algorithm == nullptr)
+    {
+      d->initialize();
+    }
+
+    if(d->feature_track_set_ptr == nullptr)
+    {
+      d->load_tracks();
+      if(d->feature_track_set_ptr == nullptr)
+      {
+        LOG_ERROR(main_logger, "There are no feature tracks.");
+        return EXIT_FAILURE;
+      }
+    }
+
+    if(d->sfm_constraint_ptr == nullptr)
+    {
+      d->load_sfm_constraint();
+    }
+
+    d->run_algorithm();
+
+    if(!d->write_cameras())
+    {
+      return EXIT_FAILURE;
+    }
+
+    if(!d->write_landmarks())
+    {
+      return EXIT_FAILURE;
+    }
+
+    return EXIT_SUCCESS;
+  }
+  catch (std::exception const& e)
+  {
+    LOG_ERROR(main_logger, "Exception caught: " << e.what());
+
+    return EXIT_FAILURE;
+  }
+  catch (...)
+  {
+    LOG_ERROR(main_logger, "Unknown exception caught");
+
+    return EXIT_FAILURE;
+  }
+} // run
+
+// ----------------------------------------------------------------------------
+void
+init_cameras_landmarks::
+add_command_options()
+{
+  m_cmd_options->custom_help( wrap_text( "[options]\n" ) );
+
+  m_cmd_options->add_options()
+  ( "h,help",     "Display applet usage" )
+  ( "c,config",   "Configuration file for tool", cxxopts::value<std::string>() )
+  ( "o,output-config",
+    "Output a configuration. This may be seeded with a "
+    "configuration file from -c/--config.",
+    cxxopts::value<std::string>() )
+  ( "v,video", "Input video", cxxopts::value<std::string>() )
+  ( "t,tracks", "Input tracks", cxxopts::value<std::string>() )
```

Comment:
I would suggest making tracks a positional argument since it is the required input.  The video input is optional and used to load metadata if desired, so that could stay as is.

Reply:
I think this is partly dependent on whether or not we want to have read in capabilities via the config file.  If it is required from the command line, then we cannot have it in the config file.